test/runtests.jl

```diff
@@ -1969,28 +2138,124 @@ end
     fn = Downloads.download("https://gist.githubusercontent.com/andreasnoack/9dadfb922fc45ca52bc7b2659b7c5b67/raw/f1dcdd2179b606ecbb7e23316da1ede971e7ffe8/Filip.dat")
-    filip_estimates_df = CSV.read(fn, DataFrame; skipto = 31, limit = 11, header = ["parameter", "estimate", "se"], delim = " ", ignorerepeated = true)
-    filip_data_df = CSV.read(fn, DataFrame; skipto = 61, header = ["y", "x"], delim = " ", ignorerepeated = true)
+    filip_estimates_df = CSV.read(fn, DataFrame; skipto=31, limit=11,
+        header=["parameter", "estimate", "se"], delim=" ", ignorerepeated=true)
+    filip_data_df = CSV.read(
+        fn, DataFrame; skipto=61, header=["y", "x"], delim=" ", ignorerepeated=true)
     X = [filip_data_df.x[i]^j for i in 1:length(filip_data_df.x), j in 0:10]
 
     # No weights
-    f1 = lm(X, filip_data_df.y, dropcollinear = false, method = :qr)
+    f1 = lm(X, filip_data_df.y, dropcollinear=false, method=:qr)
     @test coef(f1) ≈ filip_estimates_df.estimate rtol = 1e-7
     @test stderror(f1) ≈ filip_estimates_df.se rtol = 1e-7
 
     # Weights
-    f2 = lm(X, filip_data_df.y, dropcollinear = false, method = :qr, wts = ones(length(filip_data_df.y)))
+    f2 = lm(X, filip_data_df.y, dropcollinear=false,
+        method=:qr, wts=uweights(length(filip_data_df.y)))
     @test coef(f2) ≈ filip_estimates_df.estimate rtol = 1e-7
     @test stderror(f2) ≈ filip_estimates_df.se rtol = 1e-7
 end
 
 @testset "Non-finite deviance. Issue 417" begin
     X_fn = Downloads.download("https://github.com/JuliaStats/GLM.jl/files/6279630/x.txt")
     y_fn = Downloads.download("https://github.com/JuliaStats/GLM.jl/files/6279632/y.txt")
-    X_df = CSV.read(X_fn, DataFrame, header = false)
-    y_df = CSV.read(y_fn, DataFrame, header = false)
+    X_df = CSV.read(X_fn, DataFrame, header=false)
+    y_df = CSV.read(y_fn, DataFrame, header=false)
     df = hcat(y_df, select(X_df, Not("Column1")))
     ft = glm(@formula(Column1 ~ Column2 + Column3 + Column4), df, Gamma(), LogLink())
-    @test coef(ft) ≈ [9.648767705301294, -0.11274823562143056, 0.1907889126252095, -0.8123086879222496]
-    @test_throws DomainError glm(@formula(Column1 ~ Column2 + Column3 + Column4), df, Gamma(), LogLink(), start = fill(NaN, 4))
+    @test coef(ft) ≈
+          [9.648767705301294, -0.11274823562143056, 0.1907889126252095, -0.8123086879222496]
+    @test_throws DomainError glm(@formula(Column1 ~ Column2 + Column3 + Column4),
+        df, Gamma(), LogLink(), start=fill(NaN, 4))
+end
+
+@testset "Test weighting with collinear columns" begin
+    form.CarbC = form.Carb
+    form.awts = [0.6, 0.3, 0.6, 0.3, 0.6, 0.3]
+    form.fwts = [6, 3, 6, 3, 6, 3]
+    lm0 = fit(
+        LinearModel, @formula(OptDen ~ Carb), form; wts=aweights(form.awts), method=:qr)
+    lm1 = fit(LinearModel, @formula(OptDen ~ Carb + CarbC),
+        form; wts=aweights(form.awts), method=:qr)
+    @test coef(lm0) == coef(lm1)[1:2]
+    @test stderror(lm0) == stderror(lm1)[1:2]
```

Comment:
```suggestion
    @test stderror(lm0) == stderror(lm1)[1:2]
```

Reply:
Looks like the test on line 2181 failed because of approximation issues.

Reply:
Line 2181 still needs to be fixed. This seems to be the only failed test in the CI failed run